library/net.cpp

```diff
@@ -1100,6 +1100,8 @@ static int dnet_socket_connect(dnet_node *node, dnet_addr_socket_set &original_l
 	}
 
 	pthread_mutex_lock(&state->lock);
+
+	assert(state->succeed_count + state->failed_count <= state->total_count); // 'less' could be on timeout
```

Comment:
Those which I know of were made for impossible situations or cases where compilation issues do not allow server to work (ID size mismatch in eblob and elliptics and so on).

This is not about some bug we have just found, which can be triggered by client with different config option.

Basically, if this is something under remote control (remote config for instance), this is not allowed to crash.
If it is 110% local issue - this may assert with heavy error logs. But it has to be minimized.